Assets/Scripts/DayNightCycle/DayAndNightControl.cs

```diff
@@ -0,0 +1,257 @@
+            MoonCamera = GetComponentInChildren<Camera>();
+            //MainCamera =
+            foreach (Camera c in GameObject.FindObjectsOfType<Camera>())
+            {
+                if (c.isActiveAndEnabled)
+                {
+                    targetCam = c;
+                }
+            }
+            lightIntensity = directionalLight.intensity; //what's the current intensity of the light
+            // AFAIK procedural skybox needs this to work
+            RenderSettings.sun = directionalLight;
+
+
+            //Check if default light prefab exists, and if so, disable it
+            if (GameObject.Find("LightSet"))
+            {
+                GameObject.Find("LightSet").SetActive(false);
+            }
+
+            if (timecycle)
+            {
+                if (timecycle.overrideEnvironmentLighting)
+                {
+                    switch (timecycle.lightingOverrideMode)
+                    {
+                        case TimecycleProfile.AmbientLightingOverrideMode.Gradient:
+                            RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Trilight;
+                            break;
+                        case TimecycleProfile.AmbientLightingOverrideMode.Color:
+                            RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Flat;
+                            break;
+                    }
+                }
+                else
+                {
+                    RenderSettings.ambientMode = UnityEngine.Rendering.AmbientMode.Skybox;
+                }
+            }
+           
+
+            
+            UpdateLight(); // Initial lighting update. Without this, the lighting will look as if it's lagging when the scene just loaded
+        }
+
+
+        public void OnPhotonPlayerConnected(PhotonPlayer newPlayer)
+        {
+            Service.Settings.SyncSettings();
+            //syncs settings after waiting a few seconds(
+            StartCoroutine(ExecuteAfterTime(5));
+        }
+        IEnumerator ExecuteAfterTime(float time)
+        {
+            yield return new WaitForSeconds(time);
+
+            Service.Settings.SyncSettings();
+        }
+        private void Settings_OnTimeSettingsChanged(TimeSettings settings)
+        {
+            currentTime = (float) GameSettings.Time.currentTime;
+            DayLength = (float) GameSettings.Time.dayLength;
+            pause = (bool) GameSettings.Time.pause;
+            
+        }
+
+        private void OnDestroy()
+        {
+            Service.Settings.OnTimeSettingsChanged -= Settings_OnTimeSettingsChanged;
+            
+        }
+
+        // Update is called once per frame
+        void Update()
+        {
+           
+
+            if (MainCamera == null)
+            {
+                try
+                {
+                    MainCamera = GameObject.Find("MainCamera").GetComponent<Camera>();
```

Comment:
Will change font and remove the unused prefab reference. @WattanaGaming would you be able to look at the leaves reflecting? If it is something to do with the leaf material/texture then I think it might be another issue not related to time and lighting, but if we can fix it with lighting settings then we should.

Reply:
@FransStreicher fixed 👍

Reply:
Strange, initially not catching a nullref caused issues for the non-mc, but now it doesn't anymore, will push the changes without the try catch here.